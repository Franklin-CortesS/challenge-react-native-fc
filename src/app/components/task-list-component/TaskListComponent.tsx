
import React from "react";
import { FlatList, StyleSheet, Text, View } from 'react-native';
import { ListItem } from '@react-native-elements/base'

const TaskListComponent = (props: any) => {
    const myListEmpty = () => {
        return (
          <View style={{ alignItems: "center" }}>
          <Text style={styles.item}>No hay tareas que mostrar.</Text>
          </View>
        );
    };

    return (
        <View style={styles.container}>
          <FlatList
            data={props.tasks}
            keyExtractor={({ id }, index) => id}
            ListEmptyComponent={myListEmpty}
            renderItem={({ item }: any) => (
              <ListItem key={item.id} bottomDivider>
                <ListItem.Content style={styles.listItem}>
                  <ListItem.Title style={styles.listName}>
                    {item.task}
                  </ListItem.Title>
                </ListItem.Content>
              </ListItem>
            )}
          />
      </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1
      },
      item: {
        marginTop: 12,
        fontWeight: 'bold',
        color: "#000000"
      },
      title: {
        fontSize: 32,
      },
      listItem: {
        display: "flex"
      },
      listName: {
        display: "flex"
      }
  });

export default TaskListComponent;