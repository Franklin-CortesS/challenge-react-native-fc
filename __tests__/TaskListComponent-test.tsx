/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
import TaskListComponent from '../src/app/components/task-list-component/TaskListComponent';
import { render } from '@testing-library/react-native';
 
 describe('test for TaskList Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TaskListComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('render with empty props', () => {
    const tasks: any = [];
    const { getByText, getByPlaceholderText } = render(<TaskListComponent tasks={tasks}/>);
    expect(getByText('No hay tareas que mostrar.')).toBeTruthy();
  })

  test('render with props', () => {
    const tasks: any = [{id: 1, task: "Task 1"}];
    const { getByText, getByPlaceholderText } = render(<TaskListComponent tasks={tasks}/>);
    expect(getByText('Task 1')).toBeTruthy();
  })
})
 