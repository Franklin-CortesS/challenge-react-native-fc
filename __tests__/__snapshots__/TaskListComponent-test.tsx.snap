// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for TaskList Component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#E5E5E5",
            }
          }
        >
          No hay tareas que mostrar.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
